homeassistant/components/alarm_control_panel/mqtt.py

```diff
@@ -54,15 +56,21 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
         config.get(CONF_PAYLOAD_DISARM),
         config.get(CONF_PAYLOAD_ARM_HOME),
         config.get(CONF_PAYLOAD_ARM_AWAY),
-        config.get(CONF_CODE))])
+        config.get(CONF_CODE),
+        config.get(CONF_AVAILABILITY_TOPIC),
+        config.get(CONF_PAYLOAD_AVAILABLE),
+        config.get(CONF_PAYLOAD_NOT_AVAILABLE))])
```

Comment:
Trying it right now, my alarm is always showing as unavailable. I think you missed the `.extend(mqtt.MQTT_AVAILABILITY_SCHEMA.schema)`. Can you confirm?

Reply:
Yes, it is missing (added in #11829).

Not sure why the [unit test](https://github.com/home-assistant/home-assistant/blob/dev/tests/components/alarm_control_panel/test_mqtt.py#L195-L223) didn't complain about this, and since it didn't why it is not working for you.

Reply:
Yes I saw the unit test, this is very surprising. I will test this further but it seems it sees the `availability_topic` in my config (as it shows the alarm as unavailable) but doesn't get the value from it (which is "online"). The message is properly retained, checked with mosquitto_sub.

Here is my config:
```yaml
alarm_control_panel:
  platform: mqtt
  name: Alarm
  state_topic: home/alarm
  command_topic: home/alarm/set
  availability_topic: home/alarm/availability
```

Maybe the unit test somehow bypass some config validation? /cc @balloob

Reply:
So I've tested with this config:

```yaml
alarm_control_panel:
  platform: mqtt
  name: Alarm
  state_topic: home/alarm
  command_topic: home/alarm/set
  availability_topic: home/alarm/availability
  payload_available: online
  payload_not_available: offline
```

And it works perfectly.

My guess is that without the schema extension, the component isn't able to get default values for the config variables.

Reply:
Ah, yes. That makes sense. Without the schema the values for the payloads will both be `None`.

For the sake of completeness I'll add some extra tests for the default payloads in #11829.

Reply:
Thanks for your quick response and fixes!